windows/devices/uwb/UwbCxAdapterDdiLrp.cxx

```diff
@@ -678,6 +678,42 @@ windows::devices::uwb::ddi::lrp::From(const std::vector<UwbApplicationConfigurat
     return std::move(applicationConfigurationParametersWrapper);
 }
 
+UwbSetApplicationConfigurationParametersWrapper
+windows::devices::uwb::ddi::lrp::From(const std::vector<::uwb::protocol::fira::UwbApplicationConfigurationParameter> &uwbApplicationConfigurationParameters, const uint32_t sessionId)
+{
+    // Convert each individual parameter to its DDI wrapper.
+    std::vector<UwbApplicationConfigurationParameterWrapper> uwbApplicationConfigurationParameterWrappers{};
+    std::ranges::transform(uwbApplicationConfigurationParameters, std::back_inserter(uwbApplicationConfigurationParameterWrappers), [](const auto &uwbApplicationConfigurationParameter) {
+        return From(uwbApplicationConfigurationParameter);
+    });
+
+    // Calculate the total size required for the UWB_APP_CONFIG_PARAMS instance.
+    const std::size_t totalSize = std::accumulate(
+        std::cbegin(uwbApplicationConfigurationParameterWrappers),
+        std::cend(uwbApplicationConfigurationParameterWrappers),
+        static_cast<std::size_t>(offsetof(UWB_SET_APP_CONFIG_PARAMS, appConfigParams[0])),
+        [&](std::size_t totalSize, const auto &uwbApplicationConfigurationParameterWrapper) {
+            return totalSize + std::size(uwbApplicationConfigurationParameterWrapper);
+        });
+
+    // Instantiate a wrapper for UWB_SET_APP_CONFIG_PARAMS and fill it in.
+    UwbSetApplicationConfigurationParametersWrapper setApplicationConfigurationParametersWrapper{ totalSize };
+    UWB_SET_APP_CONFIG_PARAMS &setAppConfigParams = setApplicationConfigurationParametersWrapper.value();
+    setAppConfigParams.size = totalSize;
+    setAppConfigParams.sessionId = sessionId;
+    setAppConfigParams.appConfigParamsCount = std::size(uwbApplicationConfigurationParameterWrappers);
+
+    // Copy each of the converted UWB_APP_CONFIG_PARAM values into the UWB_SET_APP_CONFIG_PARAMS flex-array.
+    UWB_APP_CONFIG_PARAM *appConfigParam = reinterpret_cast<UWB_APP_CONFIG_PARAM *>(&setAppConfigParams.appConfigParams[0]);
+    for (auto &uwbApplicationConfigurationParameterWrapper : uwbApplicationConfigurationParameterWrappers) {
+        UWB_APP_CONFIG_PARAM &uwbAppConfigParam = uwbApplicationConfigurationParameterWrapper.value();
+        std::memcpy(appConfigParam, &uwbAppConfigParam, uwbAppConfigParam.size);
+        appConfigParam = reinterpret_cast<UWB_APP_CONFIG_PARAM *>(reinterpret_cast<uintptr_t>(appConfigParam) + uwbAppConfigParam.size);
+    }

```

Comment:
When I tested this locally, all but the first `appConfigParam` become corrupted. I can't see why just yet, but will think about it a bit more.

Reply:
Figured it out. Will push a fix to the other PR which incorporates your changes here. Basically, we can't use any standard iterator advancement logic since those rely on fixed sizes of elements in a range :(

Reply:
#168